### PR TITLE
Notebook for frequency analysis + stats refactoring

### LINKED ISSUE #528: Documentation for frequency analysis
There are useful frequency analysis functionality in xclim that are not well documented. This would be worth a dedicated notebook.

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #528 
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
* Refactor generic to move statistical functions to stats module
* Add notebook on frequency analysis
* Update cftime minimum version requirement to 1.1.0 (daysinmonth required by sdba notebook)

* **Does this PR introduce a breaking change?** 
Yes, imports for statistical function from `generic` will fail. Should we add a deprecation warning ?

* **Other information**:

### CODE REVIEW

**Comment by huard** [comment]
@sbiner Voir https://xclim--576.org.readthedocs.build/en/576/notebooks/frequency_analysis.html Est-ce compréhensible ?

File: docs/notebooks/frequency_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Compute the design value\n",
+    "frequency_analysis(pr, t=20, dist=\"genextreme\", mode=\"max\", freq=\"Y\", month=[5,6,7,8,9,10] )"
```
Comment (tlogan2000) [inline docs/notebooks/frequency_analysis.ipynb:80]: Is there a reason why you use 'Y' and not 'YS' like we usually do?

File: docs/notebooks/frequency_analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    "n = 50 * 366\n",
+    "start = np.datetime64(\"1950-01-01\")\n",
+    "time = start + np.timedelta64(1, \"D\") *  range(n)\n",
+    "#time = xr.cftime_range(start=\"1950-01-01\", periods=n)\n",
+    "\n",
+    "# Generate wet (1) /dry (0) days, then multiply by rain magnitude.\n",
+    "wet = bernoulli.rvs(.1, size=n)\n",
+    "intensity = gamma(a=4, loc=1, scale=6).rvs(n)\n",
+    "pr = xr.DataArray(wet * intensity, dims=(\"time\",), coords={\"time\": time}, attrs={\"units\": \"mm/d\"})\n",
+    "pr"
```
Comment (tlogan2000) [inline docs/notebooks/frequency_analysis.ipynb:49]: would `pr.plot()` be more interesting here?  I'm pretty visual

**Review by tlogan2000 — APPROVED** [review]
This seems pretty clean and understandable to me and I am not a real stats wizard so good job
Made a few tiny comments...

File: docs/notebooks/frequency_analysis.ipynb (hunk 1 above)
Comment (huard) [reply docs/notebooks/frequency_analysis.ipynb:80]: None, in this case it doesn't really matter.

File: docs/notebooks/frequency_analysis.ipynb (hunk 2 above)
Comment (huard) [reply docs/notebooks/frequency_analysis.ipynb:49]: It's a synthetic time series, so it probably doesn't look like a realy precip time series. But yeah, we should do that once we use real precip time series.

Comment (tlogan2000) [reply docs/notebooks/frequency_analysis.ipynb:49]: ok

File: docs/notebooks/frequency_analysis.ipynb [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    "# The fitting dimension is hard-coded as `time`.\n",
+    "params = fit(sub, dist=\"genextreme\")\n",
+    "params"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, the last step is to compute the percentile, or quantile, using the fitted parameters, using the `parametric_quantile` function. The function uses metadata stored in attributes of the parameters generated by `fit` to determine what distribution to use and what are the units of the quantiles. Here we need to pass the quantile [0,1], which for exceedance probabilities is just :math`1 - 1/T`."
```
[the remark this answers is not in the document]
Comment (sbiner) [reply docs/notebooks/frequency_analysis.ipynb:136]: aussi, les attributs des sorties de `parametric_quantile` et de `fa` que je vois dans le notebook me semblent erronés (p. ex. la description, ...)

File: docs/notebooks/frequency_analysis.ipynb [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "As a convenience utility, the two last steps (`fit` and `parametric_quantile`) are bundled into the `fa` function, which takes care of converting the return period into a quantile value, and renames the `quantile` output dimension to `return_period`. This dimension renaming is done to avoid name clashes with the `quantile` method. Indeed, it's often necessary when analysing large ensembles, or probabilistic samples, to compute the quantiles of the quantiles, which will cause `xarray` to raise an error."
+   ]
```
Comment (sbiner) [inline docs/notebooks/frequency_analysis.ipynb:161]: peut-être expliquer pourquoi `mode="high"`

**Review by sbiner — COMMENTED** [review]
Oui, c'est compréhensible. C'est un bon tuto et j'aurais eu intérêt à le voir avant le projet du port de MTL.
J'ai mis qqs commentaires et ajouté à un commentaire de @tlogan2000

**Pull request merged**